CMark.hsc

```diff
@@ -255,7 +342,16 @@ ptrToNodeType ptr = do
          -> return SOFTBREAK
        #const CMARK_NODE_LINEBREAK
          -> return LINEBREAK
-       _ -> error "Unknown node type"
+       _ -> if nodeType == fromIntegral (Unsafe.unsafePerformIO $ peek c_CMARK_NODE_STRIKETHROUGH) then
+              return STRIKETHROUGH
+            else if nodeType == fromIntegral (Unsafe.unsafePerformIO $ peek c_CMARK_NODE_TABLE) then
+              TABLE <$> alignments
+            else if nodeType == fromIntegral (Unsafe.unsafePerformIO $ peek c_CMARK_NODE_TABLE_ROW) then
+              return TABLE_ROW
+            else if nodeType == fromIntegral (Unsafe.unsafePerformIO $ peek c_CMARK_NODE_TABLE_CELL) then
+              return TABLE_CELL
```

Comment:
UGHHHH. THESE ARE ALL SO BAD. ????????

I don't know what to do. I think I should probably just bite the bullet _upstream_ and bake these into `cmark-gfm` as normal constants, honestly. That would make this (and any other porting efforts) so much easier.